client/app/scripts/selectors/nodes-chart.js

```diff
@@ -0,0 +1,32 @@
+import { createSelector } from 'reselect';
+import { Map as makeMap } from 'immutable';
+
+
+const allNodesSelector = state => state.get('nodes');
+
+export const nodesSelector = createSelector(
+  [
+    allNodesSelector,
+  ],
+  allNodes => allNodes.filter(node => !node.get('filtered'))
```

Comment:
Ok, I misunderstood your comment, thinking that every element that passes the filter gets recreated, but if only the returned collection is new, this actually isn't be a problem as long as we use this selector as input to *map selectors*, which implement per-element caching. There are actually a couple of places in the code where this is the case, so I won't put a special comment here.